### PR TITLE
Closes #106 #120

### LINKED ISSUE #106: Documentation:  Roxygen headers fill in values, examples or set to internal
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

Any non-user facing functions should be set to internal and not be shown on the packagdown site.

If they are shown on the site, then they should be well-documented with examples, filled out header and arguments, etcs.

### PR DESCRIPTION
**Pull request by thomasp-b**
### Thank you for your Pull Request! 

We have developed a Pull Request template to aid you and our reviewers.  Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the logrx codebase remains robust and consistent.  

### The spirit of logrx

While many packages to facilitate the logging of code already exist in the R ecosystem, it is hard to find a solution that works well for clinical programming applications. Many logging implementations are more implicit and rely on user input to create the log for the execution of a script. While this is useful for logging specific events of an application, in clinical programming a log has a set purpose.

logrx is built around the concept of creating a log for the execution of an R script that provides an overview of what happened as well as the environment that it happened in. We set out to create a flexible logging utility that could provide the necessary information to anyone reviewing the code execution so they can recreate the execution environment and run the code for themselves. Please make sure your Pull Request meets this **spirit of logrx**.

Please check off each taskbox as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `dev` branch until you have checked off each task.

- [x] The spirit of logrx is met in your Pull Request
- [x] Check that your Pull Request is targeting the dev branch, Pull Requests to master should use the [Release Pull Request Template](https://github.com/atorus-research/logrx/blob/main/.github/PULL_REQUEST_TEMPLATE/release.md)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests
- [x] Creation/updates to relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Address any updates needed for vignettes and/or templates
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: R/axecute.R [hunk 1]
```diff
@@ -1,21 +1,26 @@
 ### Functions to setup and run code using logrx
 
-#' Create a log and run a file
+#' Creation of a log and execution of a file
```
Comment (bms63) [inline R/axecute.R:3]:
```suggestion
#' Creation of a log and axecution of a file
```

File: R/log.R [hunk 2]
```diff
@@ -1,11 +1,12 @@
 ### Functions to initialise, configure, cleanup, and write the log.rx environment
 
-#' Initialises the log.rx environment
+#' Initialisation of the log.rx environment
+#'
+#' `log_init()` initialises the log.rx environment
```
Comment (bms63) [inline R/log.R:5]: Should we have something on why we want to do this?

File: R/log.R [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 #'
 #' @return Nothing
 #' @export
 #'
```
Comment (bms63) [inline R/log.R:9]: does it make sense to have an example?

File: R/log.R [hunk 4]
```diff
@@ -82,10 +86,14 @@ log_config <- function(file = NA, log_name = NA, log_path = NA){
    set_log_element("lint_results", get_lint_results(file))
 }
 
-#' Cleans up log and does checks against elements
+#' Cleaning-up of log.rx object
```
Comment (bms63) [inline R/log.R:89]: Can we also have something on why we think we should be cleaning up the environment?

**Review by bms63 — APPROVED** [review]
Hey Thomas.  Thanks for putting this together.  I was thinking since we only have 8? functions? that we are exporting that we should maybe clarify on why we think they are important to the log scripting process and add more examples where applicable.  Maybe we can revisit before the release.

**Comment by thomasp-b** [comment]
> Hey Thomas. Thanks for putting this together. I was thinking since we only have 8? functions? that we are exporting that we should maybe clarify on why we think they are important to the log scripting process and add more examples where applicable. Maybe we can revisit before the release.

Yes, it does make sense. I'll work on articulating the why and examples before our next catch-up.

**Comment by thomasp-b** [comment]
> Hey Thomas. Thanks for putting this together. I was thinking since we only have 8? functions? that we are exporting that we should maybe clarify on why we think they are important to the log scripting process and add more examples where applicable. Maybe we can revisit before the release.

@bms63, I've added examples to axecute() and to write_log_header() however, I wasn't sure about adding examples to the log_* functions or articulating the reason for using log_* functions. In terms of examples, most of them will only show a call to a log_* function. As for articulating the reason for using the log_* functions, we could either do it in a vignette or add a sentence in the log_* functions' documentation like "As an alternative to axecuting, use log_*() to control when to initialise/clean-up/... the log.rx environment". What do you think?

**Comment by thomasp-b** [comment]
> > Hey Thomas. Thanks for putting this together. I was thinking since we only have 8? functions? that we are exporting that we should maybe clarify on why we think they are important to the log scripting process and add more examples where applicable. Maybe we can revisit before the release.
> 
> @bms63, I've added examples to axecute() and to write_log_header() however, I wasn't sure about adding examples to the log_* functions or articulating the reason for using log_* functions. In terms of examples, most of them will only show a call to a log_* function. As for articulating the reason for using the log_* functions, we could either do it in a vignette or add a sentence in the log_* functions' documentation like "As an alternative to axecuting, use log_*() to control when to initialise/clean-up/... the log.rx environment". What do you think?

@bms63, I added examples to log_* functions as we discussed in our catch-up. I had to expose the non-exported run_safely_loudly() function. Let me know what you think.

File: R/axecute.R [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
 #'
 #' @importFrom purrr map_chr
 #'
-#' @return 0 if there are no errors and 1 if any error
+#' @return 0 if there are no errors or 1 if there are any errors
 #' @export
 #'
+#' @examples
+#' dir <- tempdir()
+#' text <- 'print("Hello, Timberperson!")'
```
Comment (bms63) [inline R/axecute.R:28]:
```suggestion
#' text <- 'print("Hello, logrxperson!")'
```

**Review by bms63 — COMMENTED** [review]
I think we should move the addin down to the utilities section 
image

**Comment by bms63** [comment]
image

I think this phrase or something similar should be included in each of these functions.  90% of users are just going to use `axecute ()` but might be drawn to the `log_*` functions.  WE should probably point them back to `axecute` as well unless they want to do more serious customization. 

What do you think?

File: R/utils-approved.R [hunk 6]
```diff
@@ -1,22 +1,25 @@
 #' Build approved packages and functions tibble
 #'
-#' A utility function to help you build your approve package and functions list.
-#' This can be used by logrx to log unapproved use of packages and functions.
+#' A utility function to help you build your approved packages and functions
+#' list. This can be used by logrx to log unapproved use of packages and
+#' functions.
 #'
 #' For more details see the help vignette:
```
Comment (bms63) [inline R/utils-approved.R:7]:
```suggestion
#' For more details see the vignette:
```

**Review by bms63 — CHANGES_REQUESTED** [review]
Just a few easy discussion topics for Monday and then lets get this merged!!

**Review by bms63 — APPROVED** [review]

**Pull request merged**